src/nvim/ex_cmds2.c

```diff
@@ -1045,9 +1045,10 @@ static void profile_reset(void)
         uf->uf_tm_self      = profile_zero();
         uf->uf_tm_children  = profile_zero();
 
-        XFREE_CLEAR(uf->uf_tml_count);
-        XFREE_CLEAR(uf->uf_tml_total);
-        XFREE_CLEAR(uf->uf_tml_self);
+        for (int i = 0; i < uf->uf_lines.ga_len; i++) {
+          uf->uf_tml_count[i] = 0;
+          uf->uf_tml_total[i] = uf->uf_tml_self[i] = 0;
+        }
```

Comment:
Hmm. is this avoiding the root problem by not freeing the pointer? In https://github.com/neovim/neovim/issues/12255 it looks like `fp->uf_tml_count[i]` is NULL. So if we don't free it here, then we avoid the crash... but the profile results will be wrong?

Reply:
The change by vim-patch:8.1.0130 (https://github.com/neovim/neovim/commit/7823b357812b06da7348a5b4f5a66afae8f1f71c) needs to be changed to clear the memory at zero instead of freeing it. We missed it. Therefore, this change does not result in the wrong profile result.

There were other things that seemed to need fixing, so I fixed that too. I also added a test because I think the reason we missed this was that the test was incomplete.